src/hb-set.hh

```diff
@@ -716,27 +719,43 @@ struct hb_set_t
       return *codepoint != INVALID;
     }
 
-    page_map_t map = {get_major (*codepoint), 0};
-    unsigned int i;
-    page_map.bfind (map, &i, HB_BFIND_NOT_FOUND_STORE_CLOSEST);
-    if (i < page_map.length && page_map[i].major == map.major)
+    unsigned int major = get_major (*codepoint);
+    unsigned int i = last_page_lookup;
+
+    if (unlikely (i >= page_map.length || (page_map.arrayZ[i]).major != major))
     {
-      if (pages[page_map[i].index].next (codepoint))
+      page_map_t map = {major, 0};
```

Comment:
Add a new `cmp()` method to `page_map_t` that takes `uint32_t major` so you don't need this line.